src/decode.rs

```diff
@@ -227,15 +227,11 @@ pub fn hex_decode_with_case(
     dst: &mut [u8],
     check_case: CheckCase,
 ) -> Result<(), Error> {
-    if src.len() & 1 != 0 {
-        return Err(Error::InvalidLength(src.len()));
+    let len = dst.len().checked_mul(2).unwrap();
```

Comment:
I'd like to suggest a further refinement by wrapping an Error::Overflow return instead of a direct panic.